### PR TITLE
tools: add tools framework

### PR DESCRIPTION
**Pull request by nyirit**

### CODE REVIEW

File: scoap3/modules/tools/tasks.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            len(result['data']), tool_name, result_email))
+
+        content_type, serialized_result = serializer_function(result)
+        logger.info('Result serialized, sending email... result_data_count=%d tool_name=%s result_email=%s' % (
+            len(result['data']), tool_name, result_email))
+
+        send_result(serialized_result, content_type, [result_email], tool_name)
+
+        logger.info('Result successfully sent. tool_name=%s result_email=%s' % (tool_name, result_email))
+    except Exception:
```
Comment (ammirate) [inline scoap3/modules/tools/tasks.py]: I would put `result = tool_function(**kwargs)` and ` send_result` in twp different try/except blocks.
If the task succeeds and the email fail, we might want to communicate it (i.e. if the attachment is too big, then just provide a hint for retrieving the csv manually from somewhere)

File: scoap3/modules/tools/tasks.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    :param tool_name: name of the tool, which is used to determine the generator function.
+    :param serializer_function: serializer function
+    :param kwargs: additional kwargs passed to the tool_function
+    """
+    try:
+        logger.info('Running tool. tool_name=%s result_email=%s' % (tool_name, result_email))
+
+        tool_function = current_app.config.get('TOOL_FUNCTIONS', {}).get(tool_name)
+        if tool_function is None:
+            logger.error('Invalid tool name: %s' % tool_name)
```
Comment (ammirate) [inline scoap3/modules/tools/tasks.py]: `log.error` sends events to Sentry, this is more a `warn`

File: scoap3/modules/tools/tasks.py (hunk 1 above)
Comment (ammirate) [reply scoap3/modules/tools/tasks.py]: Of course, internally :point_up: we can also think about alternative ideas maybe

**Review by ammirate — APPROVED** [review]
Minor things, looks good overall. :+1:

File: scoap3/modules/tools/tasks.py (hunk 1 above)
Comment (nyirit) [reply scoap3/modules/tools/tasks.py]: :+1:

**Review by ammirate — APPROVED** [review]

**Pull request merged**